samples/constitutions/default/actions.js

```diff
@@ -372,6 +372,17 @@ const actions = new Map([
       function (args) {
         checkX509CertBundle(args.cert, "cert");
         checkType(args.member_data, "object?", "member_data");
+        checkType(args.recovery_role, "string?", "recovery_role");
```

Comment:
There's also a `checkEnum()` that will be slightly more precise (and prevent a much-later error, if C++ code tries to deserialise an unknown/badly-cased string). It doesn't have the `?` optional syntax, so I think it would be something like:

```suggestion
        const recovery_role = args.recovery_role;
        if (recovery_role !== undefined) {
          checkEnum(
            recovery_role,
            ["NonParticipant", "Participant", "Owner"],
            "recovery_role"
          );
        }
```

(I'm not completely sure of the semantics here - should `recovery_role` be `null` or `"NonParticipant"`?)